### DISCUSSION TITLE
How do I run a swc plugin without transformation?

### DISCUSSION THREAD

**Discussion by guru-irl** [swc-project/swc#9809] [Q&A] [answered]
Background:
1. We have an existing wasm swc plugin that does some ast tranforms
2. We want to switch our bundler to esbuild, without losing those ast transforms
3. We're trying to run the swc plugin as a pre-requisite step before running esbuild

Is there a way to run the swc plugin without transforming the code from typescript to javascript? We want the output code to retain types after the plugin is run

**Comment by kdy1** [comment, marked as answer]
https://github.com/swc-project/swc/blob/33972570045e1d43a565b52ce5226489d54d2deb/packages/types/index.ts#L677-L680

**Reply by kdy1** [reply]
Basically this option disables all transforms performed by `@swc/core`, except Wasm plugins.
